File: i2f-springboot/i2f-springboot-secure-starter/src/main/java/i2f/springboot/secure/consts/SecureConsts.java
package i2f.springboot.secure.consts;

/**
 * @author Ice2Faith
 * @date 2023/6/13 8:51
 * @desc
 */
public interface SecureConsts {
    int RSA_KEY_SIZE_1024 = 1024;
    int RSA_KEY_SIZE_2048 = 2048;

    int AES_KEY_SIZE_128 = 128;
    int AES_KEY_SIZE_192 = 192;
    int AES_KEY_SIZE_256 = 256;

    // secure sign web header
    String DEFAULT_SECURE_HEADER_NAME = "sswh";

    // secure sign web parameter
    String DEFAULT_SECURE_PARAMETER_NAME = "sswp";

    // secure sign web client asym sign
    String DEFAULT_SECURE_CLIENT_ASYM_SIGN_NAME = "sswcas";

    // 客户端秘钥对的获取策略，是否是本地生成交换策略
    boolean DEFAULT_SECURE_SWAP_ASYM_KEY = true;

    String FILTER_EXCEPTION_ATTR_KEY = "SECURE_FILTER_EXCEPTION";

    String FILTER_DECRYPT_HEADER = "SECURE_FILTER_DECRYPT";

    String FILTER_ENCRYPT_HEADER = "SECURE_FILTER_ENCRYPT";

    String FLAG_ENABLE = "true";

    String FLAG_DISABLE = "false";

    String DEFAULT_HEADER_SEPARATOR = ";";

    String SECURE_HEADER_ATTR_KEY = "SECURE_HEADER";


    // 是否是String返回值类型标记
    String STRING_RETURN_HEADER = "SECURE_RETURN_STRING";

    String ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers";

    String ASYM_KEY_FILE_NAME = "asymmetric.key";

    String ASYM_CLIENTS_KEY_FILE_NAME = "clients-asym.key";

    // 是否包含动态刷新的RSA公钥签名/公钥头
    String SECURE_DYNAMIC_KEY_HEADER = "skey";
    String SECURE_CLIENT_KEY_HEADER = "wkey";

    String SECURE_REQUIRE_RESPONSE = "SECURE_REQUIRE_RESPONSE";

    String DEFAULT_ENC_URL_PATH = "/enc/";

}

File: i2f-springboot/test-swl-starter/web/secure-web/src/secure/consts/SecureConsts.js
/**
 * 定义常量
 */
const SecureConsts = {
    RSA_KEY_SIZE_1024: () => 1024,
    RSA_KEY_SIZE_2048: () => 2048,

    AES_KEY_SIZE_128: () => 128,
    AES_KEY_SIZE_192: () => 192,
    AES_KEY_SIZE_256: () => 256,

    // 默认安全头
    DEFAULT_SECURE_HEADER_NAME: () => 'sswh',

    DEFAULT_SECURE_PARAMETER_NAME: () => 'sswp',

    DEFAULT_SECURE_CLIENT_ASYM_SIGN_NAME: () => 'sswcas',

    // 默认的秘钥对获取方式，使用本地生成交换秘钥对方式
    DEFAULT_SECURE_SWAP_ASYM_KEY: () => true,

    // 安全头的分隔符
    DEFAULT_HEADER_SEPARATOR: () => ';',

    // 动态asym公钥的响应头
    SECURE_DYNAMIC_KEY_HEADER: () => 'skey',
    SECURE_CLIENT_KEY_HEADER: () => 'wkey',

    // 安全Body加密标记头
    SECURE_DATA_HEADER: () => 'secure-data',

    // 安全Param加密标记头
    SECURE_PARAMS_HEADER: () => 'secure-param',

    // 安全URL编码标记头
    SECURE_URL_HEADER: () => 'secure-url',

    // 启用标记位
    FLAG_ENABLE: () => 'true',

    // 禁用标记位
    FLAG_DISABLE: () => 'false',

    // 默认的URL编码请求路径
    ENC_URL_PATH: () => '/enc/'
}

export default SecureConsts
